frame/oracle/src/lib.rs

```diff
@@ -790,10 +831,14 @@ pub mod pallet {
 			let mut to32 = AccountId32::as_ref(&account);
 			let address: T::AccountId = T::AccountId::decode(&mut to32).unwrap_or_default();
 
-			if prices.into_iter().any(|price| price.who == address) {
+			if prices.clone().into_iter().any(|price| price.who == address) {
 				return Err("Tx already submitted")
 			}
 
+			if prices.len() as u32 >= Self::asset_info(price_id).max_answers {
+				return Err("Max answers reached")
+			}
+
```

Comment:
If we swap the two checks we can avoid cloning can't we?

Reply:
ooo ya good catch